src/pages/postgraphile/make-wrap-resolvers-plugin-audit-and-soft-delete.md

```diff
@@ -0,0 +1,134 @@
+
+> NOTE to Benjie & the team: This example is provided with my limited knowledge as I have just started 
+> plugin development and with this disclaimer I can note that by using lodash and my solution for returning
+> the Delete mutation result to client is a hack that shows my limited knowledge ✌️😉
+>
+> ps. Please modify this page however you want to fit more generic usage of Postgraphile.    
+
+```js
+const _ = require('lodash')
+const { makeWrapResolversPlugin } = require('graphile-utils')
+
+module.exports = makeWrapResolversPlugin(
+  context => {
+    if (context.scope.isRootMutation) {
+      return { scope: context.scope }
+    }
+    return null;
+  },
+  ({ scope }) => async (resolver, user, args, context, _resolveInfo) => {
+
+    const { pgClient } = context
+
+    if (
+      scope.isPgUpdateMutationField &&
+      // check if pgFieldIntrospection contains "updated_at" column
+      _.find(_.get(scope, 'pgFieldIntrospection.attributes', []), {
+        kind: 'attribute',
+        name: 'updated_at'
+      })
+    ) {
+      // modify input.patch with updatedAt value (eg. timestamp)
+      _.set(args, 'input.patch.updatedAt', (new Date()).toISOString())
+    } else if (
+      scope.isPgCreateMutationField &&
+      // check if pgFieldIntrospection contains "created_at" column
+      _.find(_.get(scope, 'pgFieldIntrospection.attributes', []), {
+        kind: 'attribute',
+        name: 'created_at'
+      })
+    ) {
+      let objectName = _.first(_.keys(args.input))
+      _.set(args, `input.${objectName}.createdAt`, (new Date()).toISOString())
+    } else if (
+      scope.isPgDeleteMutationField &&
+      _.find(_.get(scope, 'pgFieldIntrospection.attributes', []), {
+        kind: 'attribute',
+        name: 'deleted_at'
+      })
+    ) {
+
+      // hack: reconstructing schema + table name from pgFieldIntrospection. 
+      const table = _.get(scope, 'pgFieldIntrospection.namespaceName') + '.' + _.get(scope, 'pgFieldIntrospection.name')
+      // hack: reconstructing return payload GraphqlType from table name.
+      const gqlName = '@' + _.camelCase(_.get(scope, 'pgFieldIntrospection.name'))
+      let resultPayload = {
+        data: {
+          __identifiers: [],
+        },
+      }
+
+      await pgClient.query('savepoint soft_delete');
+      try {
+        const result = await pgClient.query(`
+            update ${table} 
+            set deleted_at = now()
+            where id = $1 returning *
+        `, [args.input.id])
+        const row = _.first(result.rows)
+  
+        // hack: injecting table data for soft-deleted row to graphql result.
+        if (row) {
+          let dataObject = {}
+          _.each(row, (value, key) => {
+            if (_.isDate(value)) {
+              value = value.toISOString()
+            }
+            dataObject[_.camelCase(key)] = value
+          })
+          resultPayload.data[gqlName] = dataObject
+        }
```

Comment:
This block should be replaced with a call to `_resolveInfo.graphile.selectGraphQLResultsFromTable`; it may seem redundant to have `returning *` followed by another `select`, but trust me it isn't. Since the row still exists, fortunately we **don't** have to do the `viaTemporaryTable` workaround in the actual delete:

https://github.com/graphile/graphile-engine/blob/6e466f582879ecabfc484e3e4b91eb8e799eb6ef/packages/graphile-build-pg/src/plugins/PgMutationUpdateDeletePlugin.js#L147-L185